### PR TITLE
Depth First Search in C#

### PR DESCRIPTION
**Pull request by shreyakapoor08**
Fixes #2172 

#### Checklist:
- [x] 4 space indentation.
- [x] Coding conventions are followed.
- [x] Input is taken dynamically.
- [x] Sample Input / Output is added at the end of file.
- [x] Logic Documentation (Comments).
- [x] File names are correct.

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

#### Changes proposed in this pull request:
Code added for Depth First Search in C#

#### Languages Used:
- C#

#### Files Added:
- Depth_First_Search.cs

> We're happy to help you get this ready -- don't be afraid to ask for help.

Thanks!

### CODE REVIEW

File: Depth_First_Search/Depth_First_Search.cs [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+        bool[] traversed = new bool[vertix]; 
+  
+        DFSFunction(newv, traversed); 
+    } 
+    
+  public static void Main(String[] args) 
+    { 
+        DirectedGraph newgraph = new DirectedGraph(4); 
+  
+        newgraph.Addition_of_Edge(0, 2);
```
Comment (MastersAbh) [inline Depth_First_Search/Depth_First_Search.cs:47]: Please take dynamic inputs

Comment (shreyakapoor08) [reply Depth_First_Search/Depth_First_Search.cs]: @MastersAbh Changes done. Please check

File: Depth_First_Search/Depth_First_Search.cs [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+          newgraph.Addition_of_Edge(x, y);
+        }
+  
+        Console.WriteLine("Enter vertex which is the starting point");
+        int s = Convert.ToInt32(Console.ReadLine());
+        Console.WriteLine("Depth First Search Traversal starting from vertex " + s + " is : "); 
+        newgraph.Depth_First_Search(s); 
+    } 
+} 
+// Sample input: 0 2 0 1 2 1 2 0 3 2 2 2
```
Comment (MastersAbh) [inline Depth_First_Search/Depth_First_Search.cs:61]: what does the sample input say ?

I see first thing getting read is number of edges and the first value is 0, how can it be 0 ?

File: Depth_First_Search/Depth_First_Search.cs [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+        DFSFunction(newv, traversed); 
+    } 
+    
+  public static void Main(String[] args) 
+    { 
+        DirectedGraph newgraph = new DirectedGraph(4);
+        Console.WriteLine("Number of edges to enter");
+        int n = Convert.ToInt32(Console.ReadLine());
+        
+        for(int i=0;i<n;i++){
```
Comment (MastersAbh) [inline Depth_First_Search/Depth_First_Search.cs:49]: space after for

**Comment by shreyakapoor08** [comment]
@MastersAbh I have made the corrections.
I am not able to understand why it is showing wrong output corresponding to your sample input which you mentioned above.
I even tested it on other codes and it is showing wrong only.
Rest corrections I have made

**Comment by shreyakapoor08** [comment]
@MastersAbh Kindly change the label to "gssoc20" , I am not able to see my updated marks :)

**Pull request merged**